tsl/src/compression/compression.c

```diff
@@ -2045,6 +2045,128 @@ decompress_batches_for_insert(ChunkInsertState *cis, Chunk *chunk, TupleTableSlo
 	table_close(in_rel, NoLock);
 }
 
+/* Try to decompress the given compressed data. Used for fuzzing. */
+static int
+test_one_input_throw(const uint8 *Data, size_t Size)
+{
+	StringInfoData si = { .data = (char *) Data, .len = Size };
+
+	int algo = pq_getmsgbyte(&si);
+
+	CheckCompressedData(algo > 0 && algo < _END_COMPRESSION_ALGORITHMS);
+
+	/* Only for Gorilla for now. */
+	if (algo != COMPRESSION_ALGORITHM_GORILLA)
+	{
+		return -1;
+	}
+
+	Datum compressed_data = definitions[algo].compressed_data_recv(&si);
+	DecompressionIterator *iter =
+		definitions[algo].iterator_init_forward(compressed_data, FLOAT8OID);
+
+	for (DecompressResult r = iter->try_next(iter); !r.is_done; r = iter->try_next(iter))
+		;
+
+	return 0;
+}
+
+TS_FUNCTION_INFO_V1(ts_read_compressed_data_file);
+
+/* Read and decompress compressed data from file. Useful for fuzzing. */
+Datum
+ts_read_compressed_data_file(PG_FUNCTION_ARGS)
+{
+	char *name = PG_GETARG_CSTRING(0);
+	FILE *f = fopen(name, "r");
```

Comment:
This is reading from the cwd of the server, which might allow you to read files that you should not. You might either want to make sure that this function is not built for release builds, or have some sanity checks so that you are not reading random files in the server directory.

Reply:
Good point, I made them only accessible in debug build.